shell/platform/linux/fl_engine.cc

```diff
@@ -613,6 +659,26 @@ void fl_engine_send_mouse_pointer_event(FlEngine* self,
   self->embedder_api.SendPointerEvent(self->engine, &fl_event, 1);
 }
 
+void fl_engine_dispatch_semantics_action(FlEngine* self,
+                                         uint64_t id,
+                                         FlutterSemanticsAction action,
+                                         GBytes* data) {
+  g_return_if_fail(FL_IS_ENGINE(self));
+
+  if (self->engine == nullptr) {
+    return;
+  }
+
+  const uint8_t* action_data = nullptr;
+  size_t action_data_length = 0;
+  if (data != nullptr) {
+    g_bytes_get_data(data, &action_data_length);
```

Comment:
Presumably this should be assigning the return value to `action_data`?

(Which makes me worry about the level of coverage of the manual testing, and feel more strongly that we should have unit tests that ensure correct calls to the embedder API.)